src/alembic/versions/75857bf0c826_introduce_officer_tables.py

```diff
@@ -0,0 +1,108 @@
+    # TODO: combine all past migrations into this one
+    op.drop_table("site_user")
+    op.create_table(
+        "site_user",
+        sa.Column("computing_id", sa.String(32), nullable=False, primary_key=True),
+    )
+    op.create_unique_constraint("unique__site_user__computing_id", "site_user", ["computing_id"])
+    op.create_foreign_key(
+        "fk__site_user__user_session__computing_id", "site_user", "user_session", ["computing_id"], ["computing_id"]
+    )
+
+    op.create_table(
+        "officer_info",
+        sa.Column("legal_name", sa.String(length=128), nullable=False),
+        sa.Column("discord_id", sa.String(length=18), nullable=True),
+        sa.Column("discord_name", sa.String(length=32), nullable=True),
+        sa.Column("discord_nickname", sa.String(length=32), nullable=True),
+        sa.Column("computing_id", sa.String(length=32), nullable=False),
+        sa.Column("phone_number", sa.String(length=24), nullable=True),
+        sa.Column("github_username", sa.String(length=39), nullable=True),
+        sa.Column("google_drive_email", sa.Text(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["computing_id"], ["site_user.computing_id"], name="fk__officer_info__site_user__computing_id"
+        ),
+        sa.PrimaryKeyConstraint("computing_id", name="pk__officer_info__computing_id"),
+        sa.UniqueConstraint("computing_id", name="unique__officer_info__computing_id"),
+    )
+    op.create_table(
+        "officer_term",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("computing_id", sa.String(length=32), nullable=False),
+        sa.Column("is_active", sa.Boolean(), nullable=False),
+        sa.Column("is_complete", sa.Boolean(), nullable=False),
+        sa.Column("position", sa.String(length=128), nullable=False),
+        sa.Column("start_date", sa.DateTime(), nullable=False),
+        sa.Column("end_date", sa.DateTime(), nullable=True),
+        sa.Column("nickname", sa.String(length=128), nullable=True),
+        sa.Column("favourite_course_0", sa.String(length=32), nullable=True),
+        sa.Column("favourite_course_1", sa.String(length=32), nullable=True),
+        sa.Column("favourite_pl_0", sa.String(length=32), nullable=True),
+        sa.Column("favourite_pl_1", sa.String(length=32), nullable=True),
+        sa.Column("biography", sa.Text(), nullable=True),
+        sa.Column("photo_url", sa.Text(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["computing_id"],
+            ["site_user.computing_id"],
+            # naming convention is "fk__<current class>__<target class>__<column name>"
+            name="fk__officer_term__site_user__computing_id",
+        ),
+        sa.PrimaryKeyConstraint("id", name="pk__officer_term__id"),
+        sa.UniqueConstraint("id", name="unique__officer_term__id"),
+    )
+
+
+def downgrade() -> None:
+    op.drop_table("officer_term")
+    op.drop_table("officer_info")
+
+    op.drop_table("site_user")
```

Comment:
In the future do you plan for downgrades to be less destructive? There are op.add_column and op.drop_column

Reply:
Yes! Immediately after this feature I want to combine the first few migrations into a single one. I also want to learn the alembic api better.